pkg/aws/awscontrolplane/admit_awscontrolplane.go

```diff
@@ -154,7 +154,7 @@ func (a *Admitter) Admit(request *v1beta1.AdmissionRequest) ([]admission.PatchOp
 		b := backoff.NewMaxRetries(3, 1*time.Second)
 		err = backoff.Retry(fetch, b)
 		if err != nil {
-			a.Log("level", "debug", "message", fmt.Sprintf("No AWSCluster for AWSControlPlane %s could be found", awsControlPlaneCR.Name))
+			a.Log("level", "debug", "message", fmt.Sprintf("No AWSCluster for AWSControlPlane %s could be found: %v", awsControlPlaneCR.Name, err))
```

Comment:
That does not look right to me. When we retry and give up we should fail. What are the implications of failing or continuing regardless giving up on our desired action?

Reply:
Yeah maybe this is not the coolest way to implement it. What we want to do is this.
- try to fetch the `AWSCluster`, try few times in case it's not ready at first try
- in case we find one, apply the master attributes based on it
- in case we don't find one, use the default master attributes
Not finding an `AWSCluster` shouldn't be considered a failure because the webhook shouldn't enforce an order in which the CRs are created.

Reply:
This sounds like the perfect code comment. 😅

Reply:
You mean it would be a good idea to include what I wrote as a comment? :)

Reply:
Maybe, just to provide some context that helps others understand why certain decisions against best practices were made. The best practice we decided against here is returning the error back to the caller.

Reply:
Good point :) I'm going to add the comments.